queue_job/job.py

```diff
@@ -454,6 +465,35 @@ def __init__(self, func,
         self._eta = None
         self.eta = eta
         self.channel = channel
+        # Handle automatic.workflow.job, normally it is a normal model name.
+        # In practice only _validate_invoice_job is called through automatic.workflow.job
+        # This allows the user to select the field on account.invoice, instead of having
+        # to pick automatic.workflow.job, which doesn't have the fields of the invoice model.
+        if self.model_name == 'automatic.workflow.job':
```

Comment:
Note that I did not add a dependency on this module because this won't break this module. Inheriting the Job class in a custom module for this did not seem reasonable for this kind of change. Albeit, this is a shim.